File: petclinic-java-gradle-thymeleaf-tailwind-webpack-htmx-ts/src/main/java/dev/pollito/petclinic_java_gradle_thymeleaf_tailwind_webpack_htmx_ts/specialty/SpecialtyService.java
package dev.pollito.petclinic_java_gradle_thymeleaf_tailwind_webpack_htmx_ts.specialty;

import java.util.List;
import java.util.Map;

public interface SpecialtyService {

    List<Specialty> findAll();

    Specialty get(Integer id);

    Integer create(Specialty specialty);

    void update(Integer id, Specialty specialty);

    void delete(Integer id);

    Map<Integer, Integer> getSpecialtyValues();
}

File: petclinic-kt-gradle-thymeleaf-tailwind-webpack-htmx-ts/src/main/kotlin/dev/pollito/petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts/specialty/SpecialtyService.kt
package dev.pollito.petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts.specialty

interface SpecialtyService {

    fun findAll(): List<Specialty>

    fun `get`(id: Int): Specialty

    fun create(specialty: Specialty): Int

    fun update(id: Int, specialty: Specialty)

    fun delete(id: Int)

    fun getSpecialtyValues(): Map<Int, Int>
}
